<h3> Activity #1:</h3><h1> PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<h4>Analyst: Tricia C. Serna</h4>

pip install numpy

In [68]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [69]:
import numpy as np
import pandas as pd

<h2>Converting Datasets to Dataframes</h2>
<h3>Requirements</h3>

* DataFrame name
* File Path


In [70]:
penguins = pd.read_csv("Datasets\\penguins.csv")

In [71]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h2>Descriptive Analytics</h2>

In [72]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [73]:
penguins.describe().T

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

<h2>Checking for Null Values</h2>

In [74]:
penguins.isnull()

rowid  species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    False    False   False           False          False              False   
1    False    False   False           False          False              False   
2    False    False   False           False          False              False   
3    False    False   False            True           True               True   
4    False    False   False           False          False              False   
..     ...      ...     ...             ...            ...                ...   
339  False    False   False           False          False              False   
340  False    False   False           False          False              False   
341  False    False   False           False          False              False   
342  False    False   False           False          False              False   
343  False    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 9 columns]

<h2>Summarizing Null Values</h2>

In [75]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [76]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [77]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h2>Filtering a Datasets</h2>

In [78]:
penguins[penguins['species'] == 'Adelie']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [79]:
penguins[penguins['species'] == 'Gentoo']

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [80]:
penguins[penguins['species'] == 'Chinstrap']

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

<h2>Assigning variable Name</h2>

In [81]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [82]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df


rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [83]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [84]:
chinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 1000)]

In [85]:
chinstrap

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [86]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


In [87]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [88]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     342 non-null    float64 
 4   bill_depth_mm      342 non-null    float64 
 5   flipper_length_mm  342 non-null    float64 
 6   body_mass_g        342 non-null    float64 
 7   sex                333 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


<h2>Filling up Null Values</h2>

In [89]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [90]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [91]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [92]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [93]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [94]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [95]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [96]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [97]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [98]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [99]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [100]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [101]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [102]:
penguins['sex'].mode()

0    male
Name: sex, dtype: category
Categories (2, object): ['female', 'male']

In [103]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [104]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [105]:
penguins.columns

Index(['rowid', 'species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [106]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]

<h1>QUESTIONS AND INSIGHTS</h1>

<h3>Q1:Which island has the highest average body mass of penguins?</h3>

In [107]:
penguins.groupby("island", observed=True)["body_mass_g"].mean()

island
Biscoe       4712.956871
Dream        3712.903226
Torgersen    3715.899123
Name: body_mass_g, dtype: float64

<h3>INSIGHTS:Penguins on Biscoe Island have the highest average body mass compared to the other islands.</h3>

<h3>Q2:How many penguin species does Biscoe Island has?</h3>

In [108]:
penguins[penguins["island"] == "Biscoe"]["species"].unique()

['Adelie', 'Gentoo']
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

<h3>INSIGHTS:Biscoe Island has only two penguin species: Adelie and Gentoo.</h3>

<h3>Q3: What species of penguins has the highest body mass in Biscoe Island?</h3>

In [109]:
penguins[penguins["island"] == "Biscoe"].groupby("species",observed=True)["body_mass_g"].mean().sort_values(ascending=False)

species
Gentoo    5068.965761
Adelie    3709.659091
Name: body_mass_g, dtype: float64

<h3>INSIGHTS:On Biscoe Island, the species with the highest body mass is the Gentoo.</h3>

<h3>Q4:Which sex of the Gentoo species on Biscoe Island has the highest body mass?</h3>

In [110]:
penguins[(penguins["island"] == "Biscoe") & (penguins["species"] == "Gentoo")].groupby("sex", observed=True)["body_mass_g"].mean()

sex
female    4679.741379
male      5411.011430
Name: body_mass_g, dtype: float64

<h3>INSIGHTS:On Biscoe Island, male Gentoo penguins have a higher average body mass (5411 g) than females (4680 g).</h3>

<h3>Q5:What are the species observed?</h3>

In [111]:
penguins["species"].unique()

['Adelie', 'Gentoo', 'Chinstrap']
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

<h3>INSIGHTS:There are three observed species: Adelie, Gentoo, and Chinstrap.</h3>

<h3>Q6:Which species has the longest and shortest average bill length?</h3>

In [112]:
penguins.groupby("species", observed=True)["bill_length_mm"].mean()

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

<h3> INSIGHTS:Chinstrap penguins have the longest average bill length while Adelie penguins have the shortest average bill length. </h3>

<h3>Q7:Which species has the heaviest and lightest average body mass?</h3>

In [113]:
penguins.groupby("species", observed=True)["body_mass_g"].mean()

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h3>INSIGHT:Gentoo penguins have the heaviest average body mass, while Adelie penguins have the lightest</h3>

 <h3>Q8: Which sex has the longest and shortest flipper lengths?</h3>

In [114]:
penguins.groupby("sex", observed=True)["flipper_length_mm"].mean()

sex
female    197.363636
male      204.188997
Name: flipper_length_mm, dtype: float64

<h3>INSIGHTS: Male penguins have longer flipper lengths (204.19 mm) than females (197.36 mm)</h3>

<h3>Q9: What species has longest bill length?</h3>

In [115]:
penguins.groupby("species", observed=False)["bill_length_mm"].mean()

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

<h3>INSIGHTS: Chinstrap species has the longest bill length (48.83 mm) and Adelie species has the shortest bill length (38.83 mm)</h3>

<h3>INSIGHTS: Male penguins have longer bill lengths (45.60 mm) and have shortest bill lenght females (42.10 mm) </h3>

<h3>Q10: Which island has penguins with the deepest bill depth?</h3>

In [116]:
penguins.groupby("island", observed = True)["bill_depth_mm"].mean()

island
Biscoe       15.882447
Dream        18.344355
Torgersen    18.404830
Name: bill_depth_mm, dtype: float64

<h3>INSIGHTS: Penguins in Torgersen Island have the deepest bills, while those in Biscoe Island have the shallowest bill depths observed.</h3>

<h3>Q11: Among all the observed species, which species was observed the most?</h3>

In [117]:
penguins["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>INSIGHTS: Among all the species observed in all islands,Adelie penguins were observed the most (152) and Chistrap penguins were observed the least(68).</h3>

<h2>Correlation Analysis</h2>
<h4>Using Pandas Pearson R</h4>

<h3> Q12:Is there a correlation between bill length and bill depth of penguins?</h3>

In [118]:
peng_cor = penguins["bill_depth_mm"].corr(penguins["bill_length_mm"])

In [119]:
peng_cor

np.float64(-0.23505287035553285)

<h3>INSIGHTS: There is NO correlation between penguins bill length and bill depth</h3>

<h3>Is there a correlation between bill length and body mass of penguins? </h3>

In [120]:
bill_body_corr=penguins["bill_length_mm"].corr(penguins["body_mass_g"])

In [121]:
bill_body_corr

np.float64(0.5951098244376304)

<h3>INSIGHTS: There is a correlation between flipper lenght and body mass of penguins.</h3>

<h3> Data Aggregation</h3>

In [122]:
peng_info = penguins.groupby(['species','island','sex'],observed=True).size()

In [123]:
peng_info

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Dream      female    34
                      male      34
Gentoo     Biscoe     female    58
                      male      66
dtype: int64

<h2>Categorical to Numerical Correlation</h2>
<h4>Using One way Anova</h4>
<h4>Library: Scientific Python (SciPy)</h4>

In [124]:
from scipy.stats import f_oneway

In [125]:
male = penguins[penguins['sex']=='male']

In [126]:
female = penguins[penguins['sex']=='female']


In [127]:
male = penguins[penguins['sex']=='male']['body_mass_g']

In [128]:
female = penguins[penguins['sex']=='female']['body_mass_g']

In [129]:
f_oneway(penguins[penguins['sex']=='male']['body_mass_g'],penguins[penguins['sex']=='female']['body_mass_g'])

F_onewayResult(statistic=np.float64(68.38476171974526), pvalue=np.float64(3.0226600487050246e-15))

In [130]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]